rclpy/src/rclpy/_rclpy.c

```diff
@@ -402,6 +485,11 @@ rclpy_init(PyObject * Py_UNUSED(self), PyObject * args)
   }
   int num_args = (int)pysize_num_args;
 
+  rcl_context_t * context = (rcl_context_t *)PyCapsule_GetPointer(pycontext, "rcl_context_t");
+  if (NULL == context) {
+    return NULL;
```

Comment:
Need to `Py_DECREF(pyargs)` before returning or move this above line 476

Reply:
Does this need to be decref'ed on 484 (just above too)?

Reply:
Yeah, it should be decref'd there too

Reply:
According to this, `PyArg_ParseTuple` returns borrowed refs and does not need to be decref'ed:

https://grokbase.com/t/python/python-list/0033005faw/pyarg-parsetuple-py-decref

I can't find anything definitive in the Python docs for `PyArg_ParseTuple` for `PyObject` entries.

Reply:
Oh but that's really old... I'll keep looking.

Reply:
> According to this, PyArg_ParseTuple returns borrowed refs and does not need to be decref'ed:

Correct. After that on line 476 `pyargs` is set to the return of `PySequence_List()` which returns a new reference that does need to be decref'd.

Reply:
So, there's this:

> Note that any Python object references which are provided to the caller are borrowed references; do not decrement their reference count!
-- https://docs.python.org/3/c-api/arg.html#other-objects

So I'm not sure decrefing this is right. In fact, what I've been seeing is examples with `PyArgs_ParseTuple` that do not decref and even warnings that decrefing borrowed references is dangerous.

Perhaps I'm misunderstand? @sloretz what do you think?

Reply:
So `PySequence_List` returns a _new reference_, which does need decref, I'll make sure that's happening correctly.